spm/spm_server.c

```diff
@@ -153,16 +157,23 @@ void psa_get(psa_signal_t signum, psa_msg_t *msg)
     msg->type = active_msg->type;
     msg->handle = curr_partition->msg_handle;
     msg->rhandle = active_msg->channel->rhandle;
-    msg->size = active_msg->tx_size;
-    msg->response_size = active_msg->rx_size;
+    msg->response_size = active_msg->out_vec[0].iov_len;
+
+    for (size_t i = 0; i < PSA_MAX_INVEC_LEN; i++) {
+        msg->size[i] = active_msg->in_vec[i].iov_len;
+    }
 }
 
-size_t psa_read(psa_handle_t msg_handle, size_t offset, void *buffer, size_t len)
+size_t psa_read(psa_handle_t msg_handle, uint32_t iovec_idx, void *buf, size_t num_bytes)
```

Comment:
Fyi, normally an API change like this would push the PR to the next minor release.

Reply:
agreed

Reply:
PSA spec is still changing and we follow - that is one of the reasons why this code resides in a feature branch.